tests/src/test/scala/common/rest/WskRest.scala

```diff
@@ -0,0 +1,1482 @@
+    }
+
+    val result = invokeAction(
+      name = "apimgmt/getApi",
+      parameters = parms,
+      blocking = true,
+      result = true,
+      expectedExitCode = OK.intValue)(wp)
+    result
+  }
+
+  /**
+   * Delete an entire API or a subset of API endpoints. Parameters mirror those available in the REST.
+   *
+   * @param expectedExitCode (optional) the expected exit code for the command
+   * if the code is anything but DONTCARE_EXIT, assert the code is as expected
+   */
+  override def delete(basepathOrApiName: String,
+                      relpath: Option[String] = None,
+                      operation: Option[String] = None,
+                      expectedExitCode: Int = SUCCESS_EXIT,
+                      cliCfgFile: Option[String] = None)(implicit wp: WskProps): RestResult = {
+    val parms = Map[String, JsValue]() ++ { Map("__ow_user" -> wp.namespace.toJson) } ++ {
+      Map("basepath" -> basepathOrApiName.toJson)
+    } ++ {
+      relpath map { r =>
+        Map("relpath" -> r.toJson)
+      } getOrElse Map[String, JsValue]()
+    } ++ {
+      operation map { o =>
+        Map("operation" -> o.toJson)
+      } getOrElse Map[String, JsValue]()
+    } ++ {
+      Map("accesstoken" -> wp.authKey.toJson)
+    } ++ {
+      Map("spaceguid" -> wp.authKey.split(":")(0).toJson)
+    }
+
+    val rr = invokeAction(
+      name = "apimgmt/deleteApi",
+      parameters = parms,
+      blocking = true,
+      result = true,
+      expectedExitCode = expectedExitCode)(wp)
+    return rr
+  }
+
+  def getApi(basepathOrApiName: String, params: Map[String, String] = null, expectedExitCode: Int = OK.intValue)(
+    implicit wp: WskProps): RestResult = {
+    val whiskUrl = Uri(s"http://${WhiskProperties.getBaseControllerHost()}:9001")
+    val path = Path(s"/api/${wp.authKey.split(":")(0)}$basepathOrApiName/path")
+    val resp = getWhiskEntity(path, params, whiskUrl = whiskUrl).futureValue
+    val result = new RestResult(resp.status, getRespData(resp))
+    result
+  }
+}
+
+class RunWskRestCmd() extends FlatSpec with RunWskCmd with Matchers with ScalaFutures with WskActorSystem {
+
+  implicit val config = PatienceConfig(10 seconds, 0 milliseconds)
+  implicit val materializer = ActorMaterializer()
+  val whiskRestUrl = Uri(s"http://${WhiskProperties.getBaseControllerAddress()}")
+  val basePath = Path("/api/v1")
+
+  def getExt(filePath: String)(implicit wp: WskProps) = {
+    val sep = "."
+    if (filePath.contains(sep)) filePath.substring(filePath.lastIndexOf(sep), filePath.length())
+    else null
+  }
+
+  def request(method: HttpMethod, uri: Uri, body: String = "", creds: BasicHttpCredentials): Future[HttpResponse] = {
+    val entity = HttpEntity(ContentTypes.`application/json`, body)
+    val request = HttpRequest(method, uri, List(Authorization(creds)), entity = entity)
+    Http().singleRequest(request)
+  }
+  def createEntity(path: Path, body: String, params: Map[String, String] = null, whiskUrl: Uri = whiskRestUrl)(
```

Comment:
createEntity, postEntityParam, postEntity, deleteEntity, getWhiskEntity seems like they call all be rolled into one function that takes an HTTP verb like PUT, DELETE, GET, POST.

Reply:
done.